js/app.js

```diff
@@ -0,0 +1,110 @@
+const drawGame = {
+    appName: 'Island Race Car',
+    author: 'Guillermo Pérez',
+    version: '1.0.0',
+    license: undefined,
+    size: { 
+        w: undefined, 
+        h: undefined 
+    },
+    canvas: document.querySelector('#canvas'),
+    ctx: undefined,
+    car: undefined,
+    arr: [],
+    framesIndex: 0,
+    init() {
+        this.setContext();
+        this.setSize();
+        this.createCar();
+        this.drawAll();
+        this.setEventHandlers();
+    },
+    setContext() {
+        this.ctx = this.canvas.getContext('2d');
+    },
+    setSize() {
+        this.size = {
+            w: window.innerWidth,
+            h: window.innerHeight
+        }
+        this.canvas.setAttribute('width', this.size.w / 2);
+        this.canvas.setAttribute('height', this.size.h);
+    },
+    drawRectangle() {
+        this.ctx.fillStyle = 'grey';
+        this.ctx.fillRect(50, 0, this.canvas.width - 100, this.canvas.height);
+    },
+    drawRegularLines() {
+        this.ctx.lineWidth = 20;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.moveTo(80, 0);
+        this.ctx.lineTo(80, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+
+        this.ctx.lineWidth = 20;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.moveTo(690, 0);
+        this.ctx.lineTo(690, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+    },
+    drawDashedLines() {
+        this.ctx.lineWidth = 10;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.setLineDash([60, 20]);
+        this.ctx.moveTo(370, 0);
+        this.ctx.lineTo(370, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+        this.ctx.setLineDash([0, 0]);
+    },
+    pressStart() {
+        this.init();
+        this.collision();
+        this.framesIndex;
+        this.framesIndex % 50 === 0 ? this.createObstacles() : null;
+        setInterval(() => {
+            this.arr.forEach(elm => {
+                elm.move();
+                elm.draw();
+            });
+        }, 50);
+    },
```

Comment:
- Aquí al llamar al método `this.collision()` salta error porque no le pasaríamos ningún parámetro. Lo ideal sería meterlo dentro del intervalo para que lo continuamente y dentro del `forEach` de nuestro array. Así, si nuestro array está lleno de obstáculos, para cada uno de ellos comprobará la colisión con el coche, pasándole el `elm` (es decir, el obstáculo) como argumento:

```suggestion
    pressStart() {
        this.init();
        this.framesIndex;
        this.framesIndex % 50 === 0 ? this.createObstacles() : null;
        setInterval(() => {
            this.arr.forEach(elm => {
                elm.move();
                elm.draw();
                this.collision(elm);
            });
        }, 50);
    }
```